crates/red_knot_python_semantic/resources/mdtest/type_properties/is_disjoint_from.md

```diff
@@ -0,0 +1,39 @@
+# Tests for disjointness
+
+If two types can be disjoint, it means that it is known that no possible runtime object could ever
+inhabit both types simultaneously.
+
+TODO: Most of our disjointness tests are still Rust tests; they should be moved to this file.
+
+## Instance types versus `type[T]` types
+
+An instance type is disjoint from a `type[T]` type if the instance type is `@final` and the
+associated class of the instance type is not a subclass of `T`'s metaclass.
```

Comment:
"associated class" doesn't seem like a well-defined term to me. Do we really just mean "the class of the instance type"?

Reply:
Both kinda feel like word salad to me, but I can go with "the class of the instance type" if you find it clearer 😆

Reply:
I do :) 
```suggestion
class of the instance type is not a subclass of `T`'s metaclass.
```